cabalscan/src/CabalScan/RuleGenerator.hs

```diff
@@ -259,8 +260,18 @@ findModulePaths componentName cabalFilePath hsSourceDirs moduleNames = do
       maybe raiseError return maybePath
 
 depPackageNames :: Cabal.BuildInfo -> [Text]
-depPackageNames =
-  map (pkgNameToText . Cabal.depPkgName) . Cabal.targetBuildDepends
+depPackageNames = concatMap depNames . Cabal.targetBuildDepends
+    where
+      depNames :: Cabal.Dependency -> [Text]
+      depNames dep =
+        let
+          pkgName :: Text
+          pkgName = pkgNameToText $ Cabal.depPkgName dep
+          identifierOf :: Cabal.LibraryName -> Text
+          identifierOf (Cabal.LSubLibName name) = pkgName <> ":" <> Text.pack (Cabal.unUnqualComponentName name)
+          identifierOf _ = pkgName
+        in
+          map identifierOf $ toList $ Cabal.depLibraries dep
```

Comment:
I'd qualify this use of `toList` as `Set.toList` to save the trouble of visiting the import list to guess what it does.